lib/pbench/agent/tool_data_sink.py

```diff
@@ -134,49 +171,58 @@ def terminate(self):
+                "Failed to terminate expected collector process: '%s'", exc
+            )
+            return 0
         return 1
 
 
 class PromCollector(BaseCollector):
+    """Persistent tool data collector for tools compatible with Prometheus"""
+
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
         self.volume = self.tool_group_dir / "prometheus"
 
     def launch(self):
-
-        if self.host_tools_dict:
+        if find_executable("podman"):
             self.abort_launch = False
         else:
+            self.logger.error(
+                "Podman is not installed on this system (required by some registered tools, aborting launch)"
+            )
             return 0
 
-        config = open("prometheus.yml", "w")
-
-        config.write("global:\n  scrape_interval: 1s\n  evaluation_interval: 1s\n\n")
-        # config.write("alerting:\n  alertmanagers:\n  - static_configs:\n    - targets:\n\nrule_files:\n\n")
-        config.write(
-            "scrape_configs:\n  - job_name: 'prometheus'\n    static_configs:\n    - targets: ['localhost:9090']\n\n"
-        )
-
-        for host in self.host_tools_dict:
-            if host.startswith("local"):
-                s = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
-                s.connect(("8.8.8.8", 80))
-                host_ip = str(s.getsockname()[0])
-                s.close()
+        with open("prometheus.yml", "w") as config:
+            tool_context = []
+            for host in self.host_tools_dict:
+                # Used to get ip_addr of unnamed hosts like 'localhost.localdomain'
+                if host.startswith("local"):
+                    s = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
+                    s.connect(("8.8.8.8", 80))
+                    host_ip = str(s.getsockname()[0])
+                    s.close()
+                else:
+                    host_ip = host
+
+                for tool in self.host_tools_dict[host]:
+                    tool_dict = {}
+                    port = self.tool_metadata.getProperties(tool)["port"]
+                    tool_dict["hostname"] = host_ip + "_" + tool
+                    tool_dict["hostport"] = host_ip + ":" + port
+                    tool_context.append(tool_dict)
+            if tool_context:
+                tool_context = {"tools": tool_context}
+                yml = _create_from_template("prometheus.yml", tool_context, self.logger)
+                config.write(yml)
             else:
-                host_ip = host
-
-            for tool in self.host_tools_dict[host]:
-                port = self.tool_metadata.getProperties(tool)["port"]
-                config.write(
-                    "  - job_name: '{}_{}'\n    static_configs:\n    - targets: ['{}:{}']\n\n".format(
-                        host_ip, tool, host_ip, port
-                    )
-                )
-
-        config.close()
+                self.abort_launch = True
```

Comment:
Can you provide a more detailed description of why we need `abort_launch`?  It seems like it is because the `terminate` method needs to know if it should take action or not.  Why not just check for `self.run` is not `None`?